cmd/configure_raid.go

```diff
@@ -25,35 +25,53 @@ var configureRaidCmd = &cobra.Command{
 				logger.Fatalw("failed to create raid array", "err", err, "array", raidArray, "output", out)
 			}
 		} else {
-			var blockDeviceIDs []int
+			raidType := GetString(cmd, "raid-type")
+			if raidType == "" {
+				raidType = "linuxsw"
+			}
 
-			for _, d := range GetStringSlice(cmd, "devices") {
-				intBlockDevice, err := strconv.Atoi(d)
+			raidArray := model.RaidArray{
+				Name:  GetString(cmd, "name"),
+				Level: GetString(cmd, "raid-level"),
+			}
+
+			switch raidType {
+			case "linuxsw":
+				blockDeviceFiles := GetStringSlice(cmd, "devices")
```

Comment:
Similar to a previous comment - create small functions - `CreateLinuxSoftwareRaid()` or similar to call into from the Command function, so that this method only deals with flag validation/parsing.